pkgs/tools/backup/tarsnapper/default.nix

```diff
@@ -1,25 +1,44 @@
-{ python3Packages, fetchFromGitHub , tarsnap }:
+{ lib
+, python3Packages
+, fetchFromGitHub
+, tarsnap
+}:
 
 python3Packages.buildPythonApplication rec {
-  name = "tarsnapper-${version}";
+  pname = "tarsnapper";
   version = "0.4";
 
   src = fetchFromGitHub {
     owner = "miracle2k";
-    repo = "tarsnapper";
+    repo = pname;
     rev = version;
     sha256 = "03db49188f4v1946c8mqqj30ah10x68hbg3a58js0syai32v12pm";
   };
 
-  checkInputs = with python3Packages; [ nose pytest ];
+  propagatedBuildInputs = with python3Packages; [
+    pyyaml
+    python-dateutil
+    pexpect
+  ];
 
-  checkPhase = ''
-    py.test .
-  '';
+  checkInputs = with python3Packages; [
+    nose
```

Comment:
Are the tests executed with nose upstream?

Reply:
They are. I removed `pytestCheckHook`.